src/unix/linux_like/linux/mod.rs

```diff
@@ -792,6 +792,19 @@ s_no_extra_traits! {
         pub ifr_ifru: ::sockaddr,
     }
 
+    pub union __c_anonymous_ifc_ifcu {
```

Comment:
since it's a union, it needs to go the `s_extra_no_traits` part, then implementing Eq,PartialEq, ::hash traits. Also for unions, the `#[cfg(libc_union)]` flag needs to be taken in account. You can have a look at the [FreeBSD's implementation](https://github.com/rust-lang/libc/blob/df3f7c1fa899839d4cbd4e969ce9b4d508e6d4d0/src/unix/bsd/freebsdlike/freebsd/mod.rs#L1506) for inspiration.